spec/controllers/assignment_invitations_controller_spec.rb

```diff
@@ -56,5 +58,33 @@
       expect(assignment.assignment_repos.count).to eql(1)
       expect(user.repo_accesses.count).to eql(1)
     end
+
+    context 'github repository creation fails' do
+      before do
+        allow_any_instance_of(AssignmentRepo)
+          .to receive(:create_github_repository)
+          .and_raise(GitHub::Error)
+      end
+
+      it 'does not create a an assignment repo record' do
+        patch :accept_invitation, id: invitation.key
+
+        expect(assignment.assignment_repos.count).to eq(0)
+      end
+    end
+
+    context 'github import fails' do
+      before do
+        allow_any_instance_of(GitHubRepository)
+          .to receive(:get_starter_code_from)
+          .and_raise(GitHub::Error)
+      end
+
+      it 'removes the repository on GitHub' do
+        patch :accept_invitation, id: invitation.key
+
+        assert_requested :delete, github_url('/repositories/43894381')
```

Comment:
This controller test is making sure that if the API call to GitHub Importer fails that we delete the repository we created on GitHub so we don't leave the user in an inconsistent state (a repository on GitHub without a corresponding record in Classroom).

In this case, the repository on GitHub was successfully completed and a non-persisted `ActiveRecord` object has the `github_repository_id` in one of it's attributes, but because this test is in a different process, I don't think I have any way to access that ID until it has been persisted.

Since we're using VCR, I ran the tests once and then pulled the ID out of the cassette and hard coded it here so that I can safely assert that the correct `DELETE` request is being made.

I'd really love to get rid of this hard coded id so that if someone needs to make a new cassette in the future they won't have to know to run it and then grab the ID manually and update this.

Anyone have any ideas on how I can improve this?

cc/ @jonmagic